File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.py
def solution(topping):
    r=0
    tp1,tp2=dict(),dict()
    for t in topping:
        if t not in tp1:
            tp1[t]=1
        else:
            tp1[t]+=1
    for t in topping:
        tp1[t]-=1
        if t not in tp2:
            tp2[t]=1
        else:
            tp2[t]+=1;
        if tp1[t]==0:
            del tp1[t]
        if len(tp1)==len(tp2):
            r+=1
    return r

File: 프로그래머스/2/132265. 롤케이크 자르기/롤케이크 자르기.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<int> topping) {
    int r=0;
    map<int,int> tp1,tp2;
    for(int i=0;i<topping.size();i++) {
        tp1[topping[i]]++;
    }
    for(int i=0;i<topping.size();i++) {
        tp1[topping[i]]--;
        tp2[topping[i]]++;
        if(tp1[topping[i]]==0)
            tp1.erase(topping[i]);
        if(tp1.size()==tp2.size())
            r+=1;
    }
    return r;
}